src/ios/ios.py

```diff
@@ -0,0 +1,35 @@
+import getpass
+
+class IOs:
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def prompt(self, prompt="", hide_input=False):
+        return self._prompt(prompt, hide_input)
+
+    def _prompt(self, prompt="", hide_input=False):
+        while True:
+            try:
+                value = None
+                if hide_input:
+                    value = getpass.getpass(prompt)
+                else:
+                    value = input(prompt)
+                return value
+            except EOFError:
+                print()
+
+    def print(self, string, obj=None, force=False):
+        return self._print(string, obj, force)
+
+    def _print(self, string, obj=None, force=False):
```

Comment:
Is there a reason for this pattern where you have a private method with the same arguments as the public method? Couldn't you just use a single public method? (same question for `_prompt`)

Reply:
For ease of changing protections in subclasses (see src/notif/notif.py for an example). If you want more of an explanation, I can do that.

Reply:
Oh so I assume you don't want it exposed as public in certain subclasses?

Reply:
Exactly, it wouldn't be very appropriate of a Table subclass to allow for getting input from the user in stray methods. The public versions only really exist because I can see the case where we would want to use the input methods without a specific subclass.